### PR TITLE
improve examples' use of mesh.begin() and renewAddress()

### PR DESCRIPTION
**Pull request by 2bndy5**
This solves #203 

While reviewing all the examples, I added some comments to the Linux C++ examples and ported the work from the mesh examples in pyrf24. I can revert the changes to the python examples if people are still using python2.7 for some reason, but I'm pretty sure boost.python has phased out maintenance support for python2.

### CODE REVIEW

**Comment by TMRh20** [comment]
This commit just brought to light another issue with the examples.

```
        if(!mesh.renewAddress()){
          mesh.begin();
        }
```
I think should be?
```
        if(mesh.renewAddress() == MESH_DEFAULT_ADDRESS){
          mesh.begin();
        }
```
I guess when I wrote the examples, I was thinking it would return something else.

**Comment by 2bndy5** [comment]
I don't recall seeing that snippet anywhere. Why would you call `begin()` after `renewAddress()` fails?

But I would agree. Failure from `renewAddress()` is identified when it returns 04444.

**Comment by 2bndy5** [comment]
Wait, the docs don't mention what is returned when `renewAddress ()` fails.

I suppose it is a good sign that users haven't noticed this oversight (or that no one reported it).

**Comment by 2bndy5** [comment]
Should I also add a while loop in the child node examples that demonstrate how to `renewAddress()` on `write()` failures?
```cpp
        if (!mesh.write(&displayTimer, 'M', sizeof(displayTimer))) {
            // If a write fails, check connectivity to the mesh network
            if (!mesh.checkConnection()) {
                // The address could be refreshed per a specified timeframe or only when sequential writes fail, etc.
                printf("Renewing Address\n");
                mesh.renewAddress();
            }
            else {
                printf("Send fail, Test OK\n");
            }
        }
        else {
            printf("Send OK: %u\n", displayTimer);
        }
```

Or should we add a var `isConnected` that triggers a `while (mesh.renewAddress() == MESH_DEFAULT_ADDRESS)` in the `loop()`? This idea (to me) seems overly complex for a simple example.

**Comment by TMRh20** [comment]
> I don't recall seeing that snippet anywhere. Why would you call begin() after renewAddress() fails?

Its in all the client/child examples:

```
      // If a write fails, check connectivity to the mesh network
      if ( ! mesh.checkConnection() ) {
        //refresh the network address
        Serial.println("Renewing Address");
        if (!mesh.renewAddress()) {
          //If address renewal fails, reconfigure the radio and restart the mesh
          //This allows recovery from most if not all radio errors
          mesh.begin();
        }
      } else {
        Serial.println("Send fail, Test OK");
      }
    } else {
      Serial.print("Send OK: "); Serial.println(displayTimer);
    }
  }
```
It allows recovery from hardware failures etc. and makes the radios hot-swappable. I've swapped out faulty radios without having to reset the MCU...

> Should I also add a while loop in the child node examples that demonstrate how to renewAddress() on write() failures?

Sure.

File: examples/RF24Mesh_Example/RF24Mesh_Example.ino [hunk 1]
```diff
@@ -71,10 +79,10 @@ void loop() {
     if (!mesh.write(&displayTimer, 'M', sizeof(displayTimer))) {
 
       // If a write fails, check connectivity to the mesh network
-      if ( ! mesh.checkConnection() ) {
+      if (!mesh.checkConnection()) {
         //refresh the network address
         Serial.println("Renewing Address");
-        if (!mesh.renewAddress()) {
+        if (mesh.renewAddress() == MESH_DEFAULT_ADDRESS) {
```
Comment (2bndy5) [inline examples/RF24Mesh_Example/RF24Mesh_Example.ino:85]: This is the only instance I could find which improperly uses the return value of `renewAddress()`.

**Comment by 2bndy5** [comment]
Arduino CI seems to be having trouble with some internal command. This failure is not related to these changes. Judging from multiple runs, the changes do pass when the internal Arduino CLI command is able to fetch requested data from a server.

**Comment by 2bndy5** [comment]
I found that the `RF24::isChipConnected()` wasn't exposed in the old py wrapper. I fixed this on RF24 master branch.

**Pull request merged**